spec/support/oauth.rb

```diff
@@ -0,0 +1,77 @@
+OmniAuth.config.test_mode = true
+OmniAuth.config.mock_auth = {}
+################################################################################
+# twitter
+################################################################################
+OmniAuth.config.mock_auth[:twitter] = {
+  provider: "twitter",
+  uid:      "123456",
+  info:     {
+    name: "Example Name"
+  }
+}
+
+################################################################################
+# google
+################################################################################
+OmniAuth.config.mock_auth[:google_oauth2] = {
+  provider: "google_oauth2",
+  uid:      "223456",
+  info: {
+    name:       "Example Name",
+    email:      "email@google.com",
+    first_name: "Marcelo",
+    last_name:  "Jacobus"
+  }
+}
+
+################################################################################
+# Facebook
+################################################################################
+OmniAuth.config.mock_auth[:facebook] = {
+  provider: "facebook",
+  uid:      "2234567",
+  info:     {
+    nickname:   "jbloggs",
+    email:      "email@facebook.com",
+    name:       "Example Name facebook",
+    first_name: "Joe",
+    last_name:  "Bloggs",
+    image:      "",
+    urls:       { Facebook: "" },
+    location:   "some place",
+    verified:   true
+  }
+}
+################################################################################
+# Github
+################################################################################
+OmniAuth.config.mock_auth[:github] = {
+  provider: "github",
+  uid: "123458",
+  info: {
+    nickname:  "Nick Name",
+    email:     nil,
+    name:      "Marcelo",
+    image:     "https:       //avatars.githubusercontent.com/u/226834?",
+    urls: {
+      GitHub: "https://github.com/mjacobus",
+      Blog: "jw.org"
+    }
+  }
+}
+
+
+class OauthHelper
+  def self.providers
+    OmniAuth.config.mock_auth
+  end
+end
+
+module OauthSpecHelper
+  def stub_omniauth_with(params, origin = "/")
```

Comment:
Unused method argument - `params`. If it's necessary, use `_` or `_params` as an argument name to indicate that it won't be used. You can also write as `stub_omniauth_with(*)` if you want the method to accept any arguments but don't care about them.<br>Unused method argument - `origin`. If it's necessary, use `_` or `_origin` as an argument name to indicate that it won't be used. You can also write as `stub_omniauth_with(*)` if you want the method to accept any arguments but don't care about them.